# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1546539825 18000
#      Thu Jan 03 13:23:45 2019 -0500
# Branch stable
# Node ID b352120504bc165d41cb12ab7e0ad2d689a1c287
# Parent  abca0c2690b76d19ff597fa1c242759ad9b20856
ensure valid pointer is passed to strcpy (bug #55347)

* mxarray.in.h (mxArray::strsave): Check result of mxArray::malloc.

diff --git a/libinterp/corefcn/mxarray.in.h b/libinterp/corefcn/mxarray.in.h
--- a/libinterp/corefcn/mxarray.in.h
+++ b/libinterp/corefcn/mxarray.in.h
@@ -527,18 +527,21 @@ public:
 
   static char * strsave (const char *str)
   {
     char *retval = nullptr;
 
     if (str)
       {
         mwSize sz = sizeof (mxChar) * (strlen (str) + 1);
+
         retval = static_cast<char *> (mxArray::malloc (sz));
-        strcpy (retval, str);
+
+        if (retval)
+          strcpy (retval, str);
       }
 
     return retval;
   }
 
   static octave_value
   as_octave_value (const mxArray *ptr, bool null_is_empty = true);
 
